src/preprocess/base.py

```diff
@@ -86,3 +86,17 @@ def regrid(self,
         regridder.clean_weight_file()
 
         return ds
+
+    @staticmethod
+    def resample_time(ds: xr.Dataset,
+                      resample_length: str = 'M',
+                      upsampling: bool = False) -> xr.Dataset:
+
+        # TODO: would be nice to programmatically get upsampling / not
```

Comment:
agreed this would be cool. Could be done with the `pandas.infer_freq` function maybe? 
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.infer_freq.html